src/ComposerGpgVerify/Verify.php

```diff
@@ -0,0 +1,213 @@
+                        ),
+                        $tagSignatureOutput,
+                        $signed
+                    );
+
+                    $signed = $signed || ! self::verifySignatureValidationHasNoWarnings($tagSignatureOutput);
+
+                    if (! $signed) {
+                        $output = array_merge($output, $tagSignatureOutput);
+
+                        break;
+                    }
+                }
+            }
+
+            $packages[$packageName] = [
+                'git'       => 'dunno',
+                'signed'    => (bool) $output,
+                'signature' => implode("\n", $output),
+                'verified'  => ! $signed,
+            ];
+        }
+
+        // @TODO configuration WILL arrive, but for now I'm too lazy, so GTFO
+//        $settings = $composer->getPackage()->getExtra()['composer-gpg-verify'];
+//
+//        $allowedUnsigned  = $settings['allow-unsigned'];
+//        $allowedUntrusted = $settings['allow-untrusted'];
+
+        $allowedUnsigned  = [];
+        $allowedUntrusted = [];
+
+
+        $unSigned = array_keys(array_filter(
+            $packages,
+            function (array $package) : bool {
+                return ! $package['signed'];
+            }
+        ));
+
+        $unVerified = array_keys(array_filter(
+            $packages,
+            function (array $package) : bool {
+                return ! $package['verified'];
+            }
+        ));
+
+        $requiringSignature    = array_diff($unSigned, $allowedUnsigned);
+        $requiringVerification = array_diff($unVerified, $allowedUntrusted);
+
+        $escapes = array_values(array_unique(array_merge($requiringSignature, $requiringVerification)));
+
+        if (! $escapes) {
+            return;
+        }
+
+        // @TODO report un-trusted signatures emails, names, identifiers (full length, if possible)
+
+        throw new \RuntimeException(sprintf(
+            'The following packages need to be signed and verified, or added to exclusions: %s%s',
+            "\n",
+            implode("\n", $escapes)
+        ));
+    }
+
+    /**
+     * @TODO there must be a better way, seriously. Why does `git verify-tag` not simply produce
+     * @TODO a non-zero exit code when a signature belongs to an untrusted key?!
+     */
+    private static function verifySignatureValidationHasNoWarnings(array $output) : bool
+    {
+        return ! array_filter(
+            $output,
+            function (string $outputRow) {
+                return false !== strpos($outputRow, 'gpg: WARNING: ');
+            }
+        );
+    }
+
+    /**
+     * @param Config $config
+     *
+     *
+     * @throws \LogicException
+     * @throws \RuntimeException
+     */
+    private static function assertSourceInstallation(Config $config) : void
+    {
+        $preferredInstall = $config->get('preferred-install');
+
+        if ('source' !== $preferredInstall) {
+            throw new \LogicException(sprintf(
```

Comment:
Urgh, no, I'd rather not use the additional class at all :-\

At this stage, we're not even sure if the filesystem locations are finalised.